src/test/kotlin/org/rust/lang/core/type/RsTypeResolvingTest.kt

```diff
@@ -530,6 +530,14 @@ class RsTypeResolvingTest : RsTypificationTestBase() {
                              //^ <unknown>
     """)
 
+    fun `test resolve types with functions of the same name`() = testType("""
+        struct S;
+        fn S() {}
```

Comment:
This code cannot be compiled because both struct and function produces `S` name in value namespace